src/sentry/integrations/slack/service.py

```diff
@@ -0,0 +1,206 @@
+
+        # Get all parent notifications, which will have the message identifier to use to reply in a thread
+        parent_notifications = (
+            self._notification_message_repository.get_all_parent_notification_messages_by_filters(
+                group_ids=[activity.group.id],
+                project_ids=[activity.project.id],
+            )
+        )
+        for parent_notification in parent_notifications:
+            try:
+                self._handle_parent_notification(
+                    parent_notification=parent_notification,
+                    notification_to_send=notification_to_send,
+                    client=slack_client,
+                )
+            except Exception as err:
+                self._logger.info(
+                    "failed to send notification",
+                    exc_info=err,
+                    extra={
+                        "activity_id": activity.id,
+                        "parent_notification_id": parent_notification.id,
+                        "notification_to_send": notification_to_send,
+                        "integration_id": integration.id,
+                    },
+                )
+
+    def _handle_parent_notification(
+        self,
+        parent_notification: IssueAlertNotificationMessage,
+        notification_to_send: str,
+        client: SlackClient,
+    ) -> None:
+        # For each parent notification, we need to get the channel that the notification is replied to
+        # Get the channel by using the action uuid
+        if not parent_notification.rule_fire_history:
+            raise RuleDataError(
+                f"parent notification {parent_notification.id} does not have a rule_fire_history"
+            )
+
+        if not parent_notification.rule_action_uuid:
+            raise RuleDataError(
+                f"parent notification {parent_notification.id} does not have a rule_action_uuid"
+            )
+
+        rule: Rule = parent_notification.rule_fire_history.rule
+        rule_action = rule.get_rule_action_details_by_uuid(parent_notification.rule_action_uuid)
+        if not rule_action:
+            raise RuleDataError(
+                f"failed to find rule action {parent_notification.rule_action_uuid} for rule {rule.id}"
+            )
+
+        channel_id: str | None = rule_action.get("channel_id", None)
+        if not channel_id:
+            raise RuleDataError(
+                f"failed to get channel_id for rule {rule.id} and rule action {parent_notification.rule_action_uuid}"
+            )
+
+        block = self._slack_block_builder.get_markdown_block(text=notification_to_send)
+        payload = {"channel": channel_id, "thread_ts": parent_notification.message_identifier}
+        slack_payload = self._slack_block_builder._build_blocks(
+            block, fallback_text=notification_to_send
+        )
+        payload.update(slack_payload)
+        client.post("/chat.postMessage", data=payload, timeout=5)
```

Comment:
should we be try/excepting this and logging stuff if it fails?

Reply:
also this function feels a little fat. if we're going to use this service for other things maybe we could split some of the stuff out (later)?

Reply:
we're catching the exception at the for loop level, but I guess it wouldn't hurt to output the payload so we have more visibility, let me add that

Reply:
this particular function, like `_handle_parent_notification`?
I think these functions feel fat because of the logging, but I can try to consolidate a little more. Just don't want to preemptively over engineer anything

Reply:
done, added more logging